src/types/CharData.d.ts

```diff
@@ -0,0 +1,62 @@
+import firebase from "firebase";
+import DocumentReference = firebase.firestore.DocumentReference;
+
+export default interface CharData {
+  id?: string; // キャラクターの一意のID
```

Comment:
```suggestion
  id: string;
```
IDは必須フィールドだろうから`?`不要では

Reply:
保存時には必須だけど、使うときにはいらない場合も多い

たとえば親コンポーネントでCharDataを取得した後、
`<ChildComponent :CharData="CharData" />`
で受け取った側の子コンポーネントでは使わないとか

Reply:
実際にそのフィールドを使うかどうかじゃなくて、
インスタンスを作成するときにそれを必ず入れるかどうかで決めるのでは

Reply:
まあでもTypeScriptあんまり知らないから間違ってる可能性あるし、?つけたままでええよ

Reply:
> インスタンスを作成するときにそれを必ず入れるかどうか

けっこう入れない場面がある気がする。ワイもよくわからんがいったんありのままでおねしゃす :bow: